ARCHITECTURE.md

```diff
@@ -0,0 +1,340 @@
+# JHipster architecture
+
+## Yeoman basis
+
+Internally JHipster uses [yeoman](https://yeoman.io) as core.
+
+## File structure
```

Comment:
Maybe we can add some kind of glossary, what is a generator, what is the cli, what is jdl? Guess that might help to understand all the different constructs.